worker/src/main/java/gov/cms/ab2d/worker/service/BeneficiaryServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package gov.cms.ab2d.worker.service;
+
+import gov.cms.ab2d.common.model.Beneficiary;
+import gov.cms.ab2d.common.model.Contract;
+import gov.cms.ab2d.common.model.Coverage;
+import gov.cms.ab2d.common.repository.BeneficiaryRepository;
+import gov.cms.ab2d.common.repository.ContractRepository;
+import gov.cms.ab2d.common.repository.CoverageRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class BeneficiaryServiceImpl implements BeneficiaryService {
+
+    private final BeneficiaryRepository beneRepo;
+    private final ContractRepository contractRepo;
+    private final CoverageRepository coverageRepo;
+
+
+    /**
+     * Given a contractId and a month,
+     * search for bene information in the local db
+     *
+     * @param contractId
+     * @param month
+     * @return
+     */
+    @Override
+    public Set<String> findPatientIdsInDb(Long contractId, int month) {
+        var patientIds = coverageRepo.findActivePatientIds(contractId, month);
+        return new HashSet<>(patientIds);
```

Comment:
Should we use `LinkedHashSet`? (To keep the same order in which the IDs were inserted into the cache)

Reply:
done